alchemist-elixir-ls.el

```diff
@@ -55,14 +50,10 @@
     project-erlang-version))
 
 (defun alchemist--server-extension ()
-  (let ((extension (or alchemist-server-extension
-                       (completing-read "Choose the kind of executable that runs on this system: "
-                                        '("sh" "bat")
-                                        nil
-                                        t
-                                        ))))
-    (setq alchemist-server-extension extension)
-    extension))
+  (defvar alchemist-system-extension)
+  (cond ((eq system-type 'gnu/linux) (setq alchemist-system-extension "sh"))
```

Comment:
Is it possible at all for a system to be something other than these three? If so we may want to at least raise an error.

The pattern I've used for something like this in the past was like this: https://github.com/Trevoke/sqlup-mode.el/blob/master/sqlup-mode.el#L199

Is yours more e-lispy? Off-hand it seems weird to have the variable definition inside the function definition.

Reply:
Possible values for system-type:
```
  ‘gnu’          compiled for a GNU Hurd system.
  ‘gnu/linux’    compiled for a GNU/Linux system.
  ‘gnu/kfreebsd’ compiled for a GNU system with a FreeBSD kernel.
  ‘darwin’       compiled for Darwin (GNU-Darwin, macOS, ...).
  ‘ms-dos’       compiled as an MS-DOS application.
  ‘windows-nt’   compiled as a native W32 application.
  ‘cygwin’       compiled using the Cygwin library.
Anything else (in Emacs 26, the possibilities are: aix, berkeley-unix,
hpux, usg-unix-v) indicates some sort of Unix system.
```
The question is which operating systems does elixir-ls run on, every OS Elixir and Erlang runs on?

More e-lispy? Probably not.